lib/mongo/auth/user.rb

```diff
@@ -133,14 +137,15 @@ def hashed_password
       # @option options [ Array<String>, Array<Hash> ] roles The user roles.
       #
       # @since 2.0.0
-      def initialize(options)
+      def initialize(options, client_key = nil)
         @database = options[:database] || Database::ADMIN
         @auth_source = options[:auth_source] || @database
         @name = options[:user]
         @password = options[:password] || options[:pwd]
         @mechanism = options[:auth_mech] || :mongodb_cr
         @auth_mech_properties = options[:auth_mech_properties] || {}
         @roles = options[:roles] || []
+        @client_key = client_key
```

Comment:
See my comment in ``server/connection.rb``: You can just add the ``client_key`` to the options hash passed to the ``#initialize`` method, as all the arguments are in there anyway.